src/version-base.ts

```diff
@@ -278,7 +280,7 @@ export abstract class BaseVersioner {
         };
       })
       .sort((a, b) => {
-        return a.version.compare(b.version);
+        return b.version.compare(a.version);
```

Comment:
consider making the reverse order more explicit/obvious, it may be easy to miss in future readings of this code.
```suggestion
        // Sorted in descending order.
        return -a.version.compare(b.version);
```
(i think this is descending order.. right?)